src/app/routes/loan/shared/store/api/route-api.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Models } from '$shared';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import LOANS from '../../../../../../assets/mock-data/loans';
+// import { environment } from '$env'; // Base api url
+
+/**
+ * Route specific api stores
+ * By default they are not injected in root since they are only needed by this route
+ */
+@Injectable()
+export class RouteApiService {
+  loanDetailsLoadingState$ = new BehaviorSubject<Models.LoadingState>(Models.LoadingState.Unloaded);
+  loanDetails$ = new BehaviorSubject<Models.LoanDetails | null>(null);
+  loanDetailsErrorMessage$ = new BehaviorSubject<string>('');
+
+  getLoanDetails(loanId: number) {
+    this.loanDetailsLoadingState$.next(Models.LoadingState.Loading);
+    this.loanDetails$.next(null);
+    this.loanDetailsErrorMessage$.next('');
+
+    setTimeout(() => {
```

Comment:
May not matter since this is stub code but it can't hurt to future proof this for SSR by using a promise instead of settimeout. Node can't see settimeout.


> Promise.resolve().then(() => {...async code here})